### PR TITLE
Course management endpoints 2

### PR DESCRIPTION
**Pull request by meleciosilva**
Added endpoints for fetching all courses (list), fetching a single course (read), and deleting a course (destroy).

### CODE REVIEW

File: src/controllers/courses.controller.js [hunk 1]
```diff
@@ -27,6 +27,27 @@ function create(req, res, next) {
   });
 }
 
+// Fetch all courses
+function list(req, res, next) {
+  Course.find({}, (err, courses) => {
```
Comment (seyiogunjuyigbe) [inline src/controllers/courses.controller.js]: you may need to populate the steps here and in the read function

Comment (meleciosilva) [reply src/controllers/courses.controller.js]: Both 'list' and 'read' return the step _ids of the steps for each courses. Should they return the entire step instead?

Comment (seyiogunjuyigbe) [reply src/controllers/courses.controller.js]: Yes, populate the step fields

Comment (meleciosilva) [reply src/controllers/courses.controller.js]: Okay, I made the revisions in my latest commit "update list and read to populate steps". Let me know if any corrections are needed. Thanks!

**Review by seyiogunjuyigbe — APPROVED** [review]

**Pull request merged**